src/Arcus.Security.Core/CompositeSecretProvider.cs

```diff
@@ -364,8 +364,8 @@ private void EnsureAnySecretProvidersConfigured(string secretName)
                 _logger.LogSecurityEvent("Get Secret", new Dictionary<string, object>
                 {
                     ["SecretName"] = secretName,
-                    ["SecretProvider"] = source.SecretProvider.GetType().Name
-                }); 
+                    ["SecretProvider"] = source.Options.Name ?? source.SecretProvider.GetType().Name
```

Comment:
We could of course add both the type and the name. So that the name would become `["SecretProviderName"]` or something.